File: src/qsat.jl
function qsat(water, P, T)

    Tc = T - Tm
    if (Tc > 0.0 || water)
        es = e0 * exp(17.5043 * Tc / (241.3 + Tc))
    else
        es = e0 * exp(22.4422 * Tc / (272.186 + Tc))
    end
    Qs = epsm * es / P

    return Qs

end

File: fortran/src/QSAT.f90
!-----------------------------------------------------------------------
! Saturation specific humidity
!-----------------------------------------------------------------------
subroutine QSAT(water,P,T,Qs)

use, intrinsic :: iso_fortran_env, only: dp=>real64

use Constants, only : &
  eps,               &! Ratio of molecular weights of water and dry air
  e0,                &! Saturation vapour pressure at Tm (Pa)
  Tm                  ! Melting point (K)

implicit none

logical, intent(in) :: &
  water               ! Saturation wrt water if TRUE

real*8, intent(in) :: &
  P,                 &! Air pressure (Pa)
  T                   ! Temperature (K)

real*8, intent(out) :: &
  Qs                  ! Saturation specific humidity

real*8 :: &
  Tc,                &! Temperature (C)
  es                  ! Saturation vapour pressure (Pa)

Tc = T - Tm
if (Tc > 0 .or. water) then
  es = e0*exp(17.5043_dp*Tc / (241.3_dp + Tc))
else
  es = e0*exp(22.4422_dp*Tc / (272.186_dp + Tc))
end if
Qs = eps*es / P

end subroutine QSAT
